alf/algorithms/sac_algorithm.py

```diff
@@ -280,44 +252,56 @@ def _critic_train_step(self, exp: Experience, state: SacCriticState, actions_and
             target_critic2=target_critic2_state)
 
         info = SacCriticInfo(
-            critic1=critic1, critic2=critic2,
-            target_critic=target_critic)
+            critic1=critic1, critic2=critic2, target_critic=target_critic)
 
         return state, info
 
-    def _alpha_train_step(self, actions_and_log_probs):
-        action, log_pi = actions_and_log_probs
-        alpha_loss = self._log_alpha * tf.stop_gradient(-log_pi - self._target_entropy)
+    def _alpha_train_step(self, log_pi):
+        alpha_loss = self._log_alpha * tf.stop_gradient(-log_pi -
+                                                        self._target_entropy)
         info = SacAlphaInfo(loss=LossInfo(loss=alpha_loss, extra=alpha_loss))
         return info
 
     def train_step(self, exp: Experience, state: SacState):
-        actions_and_log_probs, share_state = self._actions_and_log_probs(exp, state.share)
```

Comment:
L266-L286 state handle is incorrect
1. use state.share.actor for actor_network and use new network_state as SacShareState

Reply:
Good catch!